src/orion/benchmark/base.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.benchmark.base` -- Base definition of Task and Assessment
+======================================================================
+
+.. module:: base
+   :platform: Unix
+   :synopsis: Base definition of Task and Assessment.
+
+"""
+
+from abc import ABC, abstractmethod
+
+
+class BaseAssess(ABC):
+    """Base class describing what an assessment can do.
+
+    Parameters
+    ----------
+    task_num : int
+       Number of experiment the assessment ask to run the corresponding task
+    kwargs : dict
+       Configurable parameters of the assessment, a particular assessment
+       implementation can have its own parameters.
+    """
+
+    def __init__(self, task_num, **kwargs):
+        self.task_number = task_num
+        self._param_names = list(kwargs.keys())
```

Comment:
Why do you keep the param names only and not the values?

Reply:
good point. I can actually simply just keep the `kwargs`.  I guess repeating changing code make you forget the original purpose :)

Reply:
Also shouldn't the kwargs (both keys and values) be part of the configuration (`def configuration` below)?

Reply:
yes, they should, they are actually already included. Only difference was it did not use kwargs directly which will be changed.